pytensor/link/numba/dispatch/basic.py

```diff
@@ -572,48 +560,43 @@ def {fn_name}({", ".join(input_names)}):
 
 
 @numba_funcify.register(Subtensor)
+@numba_funcify.register(IncSubtensor)
 @numba_funcify.register(AdvancedSubtensor1)
-def numba_funcify_Subtensor(op, node, **kwargs):
-    objmode = isinstance(op, AdvancedSubtensor)
-    if objmode:
-        warnings.warn(
-            ("Numba will use object mode to allow run " "AdvancedSubtensor."),
-            UserWarning,
-        )
-
-    subtensor_def_src = create_index_func(node, objmode=objmode)
-
-    global_env = {"np": np}
-    if objmode:
-        global_env["objmode"] = numba.objmode
-
+def numba_funcify_default_subtensor(op, node, **kwargs):
+    function_name = "subtensor"
+    if isinstance(op, SET_OR_INC_OPS):  # type: ignore
+        function_name = "setsubtensor" if op.set_instead_of_inc else "incsubtensor"
+    subtensor_def_src = create_index_func(node)
     subtensor_fn = compile_function_src(
-        subtensor_def_src, "subtensor", {**globals(), **global_env}
+        subtensor_def_src, function_name, {**globals(), "np": np}
     )
-
     return numba_njit(subtensor_fn, boundscheck=True)
 
 
-@numba_funcify.register(IncSubtensor)
-def numba_funcify_IncSubtensor(op, node, **kwargs):
-    objmode = isinstance(op, AdvancedIncSubtensor)
-    if objmode:
-        warnings.warn(
-            ("Numba will use object mode to allow run " "AdvancedIncSubtensor."),
-            UserWarning,
-        )
-
-    incsubtensor_def_src = create_index_func(node, objmode=objmode)
-
-    global_env = {"np": np}
-    if objmode:
-        global_env["objmode"] = numba.objmode
+@numba_funcify.register(AdvancedSubtensor)
+@numba_funcify.register(AdvancedIncSubtensor)
```

Comment:
Should add a check for broadcasted y values in inc operations, and for the case reported in https://github.com/numba/numba/issues/9573